### PR TITLE
[5.5] Add breaking changes related to precision to changelog

### PR DESCRIPTION
**Pull request by sisve**
Related: https://github.com/laravel/framework/pull/18962

I'll argue that this is a breaking change and should be noted as such. This is regards to migrations, which I believe is very important to be stable over long timespans.

I have several years worth of migrations, and this change will change the behavior/result of every existing already migration. Executing the migrations in 5.5 will result in a different schema than when I executed in a 5.4 installation. Thus I think this should be documented as a breaking change that needs to be highlighted for SQL Server users.

I will need to rewrite my migrations to use explicit sql commands to keep the old behavior. This is also an indication that it is a breaking change.

TL:DR; I consider this a breaking change.

### CODE REVIEW

**Comment by taylorotwell** [comment]
Is it possible for us to make this kind of thing configurable?

**Comment by sisve** [comment]
Probably, but it's way to short on time to do anything about it right now. I presume the 5.5 release is very close. I think that the short term (<1 day) solution would be to document this, and later (>1 day) expose some way to create old datetime types on SQL Server.

I think that the datetime2 implementation should have been a separate method/type on the Blueprint, just as we already do for `text`, `mediumText` and `longText` to expose mysql-specific types. But it's way to late to change that now.

**Comment by taylorotwell** [comment]
The 5.5 release is around 4 weeks away?

**Comment by sisve** [comment]
Are you sure? ;) I thought it would be released during Laracon, and the After Party starts in 40 minutes (if I understand the schedule and timezones correct) which kind of limits the window a bit. There's also the fact that 5.4 just left the usual 6 month support window. And that others also thinks that it will be released today (with vagueness implied due to timezones).

> Laravel 5.5 in July 2017 will require PHP 7+
https://twitter.com/taylorotwell/status/809767371774816256

> Laravel 5.5 is scheduled to be released in July of 2017 and will be the next major release.
https://laravel-news.com/category/laravel-5.5

> Here we go again! Currently scheduled for a July release date, we're on the verge of Laravel 5.5.
https://laracasts.com/series/whats-new-in-laravel-5-5

**Comment by taylorotwell** [comment]
No, I'm not sure why everyone keeps thinking it is releasing at Laracon. Last year it released at Laracon EU.

**Comment by sisve** [comment]
I've updated the PR and added notices about dropped support for MySQL 5.5, and SQL Server 2005 and older. This was initiated by https://github.com/laravel/framework/issues/20284

**Comment by fernandobandeira** [comment]
I think we should modify the sql to be compatible with 5.5.

Instead of this statement:
```sql
alter table "users" add "foo" datetime(0) not null
```

We could create this one:
```sql
alter table "users" add "foo" datetime not null
```

When not passing precision to the function, this way we keep the functionality and the support for 5.5

@sisve Would something like this also solve the problem of changing the datetime column to datetime2 on sqlserver? I mean we can create a datetime column if no precision is informed and a datetime2 when the parameter is available (this way all of our drivers will keep the same functionality). Also we could add another method to always create a datetime2 field.

**Comment by taylorotwell** [comment]
I agree with @fernandobandeira. @themsaid could you do this for us?

**Comment by themsaid** [comment]
Here: https://github.com/laravel/framework/pull/20465
